Ex5/Ex5.java

```diff
@@ -0,0 +1,51 @@
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class Ex5 {
+
+	public static void main(String[] args) {
+		String[] dictionary = {"art","rat","cat","car"};
+		String[] dictionary2 = {"a","b","Ab","Ac"};
+		String[] dictionary3 = {"$"};
+		
+		sortUnknownAlphabet(dictionary); // [a, t, r, c]
+		sortUnknownAlphabet(dictionary2); // [a, b, A, c]
+		sortUnknownAlphabet(dictionary3); // [$]
+	}
+	
+	public static void sortUnknownAlphabet(String[] dictionary) {
+		if (dictionary == null || dictionary.length == 0 ) {
+			throw new IllegalArgumentException();
+		}
+		
+		Graph<Character> graph = new Graph<>();
+		insertNodes(dictionary, graph);
+		insertEdges(dictionary, graph);
+		
+		System.out.println(graph.topologicSort());
```

Comment:
Please don't put this print in your method - at least return a type and keep the print statements in the main method. That is the first step to making your code testable.